warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__all_tu_in_vp.sql

```diff
@@ -0,0 +1,56 @@
+WITH
+
+services_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__services_guideline_index') }}
+),
+
+fct_observed_trips AS (
+    SELECT * FROM {{ ref('fct_observed_trips') }}
+),
+
+dim_provider_gtfs_data AS (
+    SELECT * FROM {{ ref('dim_provider_gtfs_data') }}
+),
+
+joined AS (
+    SELECT
+       idx.date,
+       idx.service_key,
+       COUNT(
+        CASE WHEN
+            f.tu_num_distinct_message_ids IS NOT NULL
+            AND
+            f.vp_num_distinct_message_ids IS NOT NULL
+        THEN 1 END ) AS vp_and_tu_present,
+       COUNT(
+        CASE WHEN
+            f.tu_num_distinct_message_ids IS NOT NULL
+        THEN 1 END) AS tu_present
+    FROM services_guideline_index AS idx
+    LEFT JOIN dim_provider_gtfs_data AS quartet
+    ON idx.service_key = quartet.service_key
+    AND idx.date BETWEEN EXTRACT(DATE FROM quartet._valid_from) AND EXTRACT(DATE FROM quartet._valid_to)
+    LEFT JOIN fct_observed_trips AS f
+    ON quartet.associated_schedule_gtfs_dataset_key = f.schedule_to_use_for_rt_validation_gtfs_dataset_key
+    AND idx.date = f.dt
+    AND f.tu_num_trips_scheduled_canceled_added > 0
+    GROUP BY 1,2
```

Comment:
nit: formatting and I think I usually like converting the date to a timestamp more than extracting the date from both `between` columns?

I might double-check with @lauriemerrell on this join in general, I don't quite understand `dim_provider_gtfs_data` fully yet